language-support/codegen-common/src/test/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReaderSpec.scala

```diff
@@ -114,6 +115,91 @@ class CodegenConfigReaderSpec extends AnyFlatSpec with Matchers {
     )
   }
 
+  it should "parse package references with >= 1 dash" in {
+    val badConfigStr = """|
+      |name: quickstart
+      |version: 1.2.3
+      |codegen:
+      |  java:
+      |    package-prefix: my.company.java.package
+      |    output-directory: path/to/output/java/directory
+      |    decoderClass: my.company.java.DecoderClass
+      |module-prefixes:
+      |  a-0.0.0: a
+      |  a-a-0.0.0: a
+      |  a-a-a-0.0.0: a
+      |  a-a-a-a-0.0.0: a
+      |  a-a-a-a-a-0.0.0: a""".stripMargin
+
+    val version = PackageVersion.assertFromString("0.0.0")
+    val prefix = "a"
+
+    val expected = Conf(
+      darFiles = Map(
+        projectRoot.resolve(".daml/dist/quickstart-1.2.3.dar") -> Some("my.company.java.package")
+      ),
+      outputDirectory = path("path/to/output/java/directory"),
+      decoderPkgAndClass = Some(("my.company.java", "DecoderClass")),
+      modulePrefixes = Seq("a", "a-a", "a-a-a", "a-a-a-a", "a-a-a-a-a").view
+        .map(x => PackageReference.NameVersion(PackageName.assertFromString(x), version) -> prefix)
+        .toMap,
+    )
+
+    codegenConf(badConfigStr, Java) shouldBe Right(
+      expected
+    )
+  }
+
+  it should "parse package references with at least one number" in {
+    val badConfigStr = """|
+      |name: quickstart
+      |version: 1.2.3
+      |codegen:
+      |  java:
+      |    package-prefix: my.company.java.package
+      |    output-directory: path/to/output/java/directory
+      |    decoderClass: my.company.java.DecoderClass
+      |module-prefixes:
+      |  a-0: a
+      |  a-0.0: a
+      |  a-0.0.0: a""".stripMargin
+
+    val name = PackageName.assertFromString("a")
+    val prefix = "a"
+
+    val expected = Conf(
+      darFiles = Map(
+        projectRoot.resolve(".daml/dist/quickstart-1.2.3.dar") -> Some("my.company.java.package")
+      ),
+      outputDirectory = path("path/to/output/java/directory"),
+      decoderPkgAndClass = Some(("my.company.java", "DecoderClass")),
+      modulePrefixes = Seq("0", "0.0", "0.0.0").view
+        .map(x => PackageReference.NameVersion(name, PackageVersion.assertFromString(x)) -> prefix)
+        .toMap,
+    )
+
+    codegenConf(badConfigStr, Java) shouldBe Right(
+      expected
+    )
+  }
+
+  it should "reject package references with no dash" in {
```

Comment:
Can we reliably make this assumption? Should we (can we?) think about separating the library name and version?

Reply:
Yes we can safely make this assumption because GHC makes this assumption and we cannot compile a package that does not follow this pattern. If we ever move away from that we could do something else but that doesn’t seem worth considering atm.